test/integration/integration_test.go

```diff
@@ -663,80 +668,120 @@ func awaitRolloutRestarts(t *testing.T, ctx context.Context,
 			}
 			return err
 		},
-		backoff.WithMaxRetries(backoff.NewConstantBackOff(time.Second*1), 60),
+		backoff.WithMaxRetries(backoff.NewConstantBackOff(time.Second*1), 360),
 	))
 }
 
+type status struct {
+	Replicas      *int32
+	ReadyReplicas int32
+}
+
+// assertRolloutRestarts asserts status and object state of each RolloutRestartTarget based on minGeneration.
+// If minGeneration==1, minimally assert each RolloutRestartTarget reached a healthy state after created
+// RolloutRestartTarget like argo.Rollout needs to reach a healthy state before we can effectively test
+// the rollout restart via a reconciler.Reconcile() - when argo.Rollout is making progress towards a healthy state
+// after created, any immediately triggered restart won't take effect
+// If minGeneration==2, assert each RolloutRestartTarget restarted, reached a healthy state, and has minGeneration
+// and appropriate restartAt.
```

Comment:
I'd suggest breaking up and formatting some of this text like below to make it easier to read. It also wouldn't hurt to wrap the lines at ~85 columns too (makes it easier to read in a diff).

```suggestion
// assertRolloutRestarts asserts status and object state of each RolloutRestartTarget based on minGeneration.
//
// If minGeneration==1, minimally assert each RolloutRestartTarget reached a healthy state after being created.
// RolloutRestartTarget's like argo.Rollout need to reach a healthy state before we can effectively test
// the rollout restart via a reconciler.Reconcile() - when argo.Rollout is making progress towards a healthy state
// after being created, any immediately triggered restart won't take effect.
//
// If minGeneration==2, assert each RolloutRestartTarget restarted, reached a healthy state, and has minGeneration
// and appropriate restartAt.
```

Reply:
Thanks a lot for reformatting these!